rust/src/measurements/make_private_lazyframe/group_by/test.rs

```diff
@@ -97,3 +100,62 @@ fn test_stable_keys_zCDP() -> Fallible<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_explicit_keys() -> Fallible<()> {
+    static N_SAMPLES: u32 = 1000;
+    static N_CANDIDATES: usize = 10;
+
+    let lf_domain = LazyFrameDomain::new(vec![
+        SeriesDomain::new("A", AtomDomain::<u32>::default()),
+        SeriesDomain::new("B", AtomDomain::<f64>::default()),
+    ])?
+    .with_margin(Margin::by(["A"]).with_max_partition_length(1))?;
+
+    let lf = df!("A" => &[0u32], "B" => &[0.0f64])?.lazy();
+    let keys = df!("A" => &(0u32..N_SAMPLES).collect::<Vec<_>>())?.lazy();
+
+    let sum_expr = col("B")
+        .fill_nan(0.0)
+        .fill_null(0.0)
+        .dp()
+        .sum((0.0, 1.0), None)
+        .alias("sum");
+    let candidates = (0..N_CANDIDATES).map(|v| v as f64).collect::<Vec<_>>();
+    let median_expr = col("B")
+        .dp()
+        .median(Series::new("".into(), candidates.clone()), None)
+        .alias("med");
+    let meas = make_private_lazyframe(
+        lf_domain,
+        SymmetricDistance,
+        MaxDivergence,
+        lf.clone()
+            .group_by(&[col("A")])
+            .agg(&[sum_expr, median_expr])
+            .join(keys, [col("A")], [col("A")], JoinType::Right.into()),
+        Some(1.),
+        None,
+    )?;
+
+    let release = meas.invoke(&lf)?.collect()?;
+    let gauss_samples: Vec<_> = release.column("sum")?.f64()?.iter().flatten().collect();
+    let gauss_samples = <[f64; 1000]>::try_from(gauss_samples).unwrap();
+
+    pub fn normal_cdf(x: f64) -> f64 {
+        (erf::erf(x / std::f64::consts::SQRT_2) + 1.0) / 2.0
+    }
+
+    kolmogorov_smirnov(gauss_samples, normal_cdf)?;
+
+    // check for uniformity of samples (all scores are matching)
+    let unif_samples: Vec<_> = release.column("med")?.f64()?.iter().flatten().collect();
+    let mut counts = [0.0; N_CANDIDATES];
+    unif_samples.iter().for_each(|&s| counts[s as usize] += 1.0);
+    chisquare(
+        counts,
+        [N_SAMPLES as f64 / (N_CANDIDATES as f64); N_CANDIDATES],
+    )?;
```

Comment:
Wouldn't mind names like `assert_kolmogorov_smirnow` and `assert_chi_square`, but not essential.